packages/react-client/src/ReactFlightClient.js

```diff
@@ -3006,84 +3046,66 @@ function flushComponentPerformance(
     }
     childTrackIdx = childResult.track;
     const childEndTime = childResult.endTime;
-    childTrackTime = childEndTime;
+    if (childEndTime > childTrackTime) {
+      childTrackTime = childEndTime;
+    }
```

Comment:
This fixes the bug where there could end up being overlapping components when they should've been bumped to a parallel track.

(I do think this type of view is not good for this since any async microtask can cause them to start overlapping and spawn too many parallel tracks.)